### ISSUE TITLE
Python 2.5 uses "<module>" instead of "?" as the default filename for code objects

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1892]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1892|
|Type|defect|
|Created|2006-07-02 23:23:52Z|

This changes the particular bytes in various tracebacks which Twisted tests against.  It may or may not be changed in Python, but I guess we'll just try to handle either name.

<details><summary>Searchable metadata</summary>

```
trac-id__1892 1892
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1151882632000000 1151882632000000
changetime__1152012467000000 1152012467000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to PenguinOfDoom|
|-|-|

Ready for review in magic-code-name-1892

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @exarkun|
|-|-|

The function in test_manhole is very clever.  However, it's also not so clever that it becomes obscure. 

The tests in test_output should be documented, but that's my problem, not yours.

Merge away.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [17478]) Merge magic-code-name-1892

Author: exarkun
Reviewer: jml
Fixes [#1892](https://github.com/twisted/twisted/issues/1892)

Change trial and conch manhole tests to account for the difference in traceback
formatting between Python <=2.4 and Python >=2.5.  In Python 2.4, tracebacks
which start at the interactive interpreter or the top-level of a module look
like this:

Traceback (most recent call last):
  File "<stdin>", line 1, in ?
ZeroDivisionError: integer division or modulo by zero

In Python 2.5, the format changed to look like this:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ZeroDivisionError: integer division or modulo by zero

These changes make the test suite work with either form.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
